docs/storing-querying/schemas-in-warehouse/index.md

```diff
@@ -1,15 +1,307 @@
 ---
-title: "How schemas translate to database types"
-date: "2021-12-30"
-sidebar_position: 100
-hide_table_of_contents: true
+title: "How schema definitions translate to the warehouse"
+sidebar_label: "Schemas in the warehouse"
+sidebar_position: 4
 ---
 
 ```mdx-code-block
 import Tabs from '@theme/Tabs';
 import TabItem from '@theme/TabItem';
 ```
 
+[Self-describing events](/docs/understanding-your-pipeline/events/index.md#self-describing-events) and [entities](/docs/understanding-your-pipeline/entities/index.md) use [schemas](/docs/understanding-your-pipeline/schemas/index.md) to define which fields should be present, and of what type (e.g. string, number). This page explains what happens to this information in the warehouse.
+
+## Location
+
+Where can you find the data carried by a self-describing event or an entity?
+
+<Tabs groupId="warehouse" queryString>
+  <TabItem value="redshift" label="Redshift and Postgres" default>
+
+Each type of self-describing event and each type of entity get their own dedicated tables. The name of such a table is composed of the schema vendor, schema name and its major version (more on versioning [later](#versioning)).
+
+:::note
+
+All characters are converted to lowercase and all symbols (like `.`) are replaced with an underscore.
+
+:::
+
+Examples:
+
+| Kind | Schema | Resulting table |
+|---|---|---|
+| Self-describing event | `com.example/button_press/jsonschema/1-0-0` | `atomic.com_example_button_press_1` |
+| Entity | `com.example/user/jsonschema/1-0-0` | `atomic.com_example_user_1` |
+
+Inside the table, there will be columns corresponding to the fields in the schema. Their types are determined according to the logic described [below](#types).
+
+:::note
+
+The name of each column is the name of the schema field converted to snake case.
+
+:::
+
+:::caution
+
+If an event or entity includes fields not defined in the schema, those fields will not be stored in the warehouse.
+
+:::
+
+For example, suppose you have the following field in the schema:
+
+```json
+"lastName": {
+  "type": "string",
+  "maxLength": 100
+}
+```
+
+It will be translated into a column called `last_name` (notice the underscore), of type `CHAR(100)`.
```

Comment:
VARCHAR not CHAR.

Actually we use CHAR in a couple of situations:
- UUID fields
- string fields where `minLength` is equal to `maxLength`
- enum fields where only one value is allowed.

In all other situations it is VARCHAR

Reply:
I think the type table below is wrong then

Reply:
Scratch that, the table is right, my mistake